x-pack/plugins/security_solution/public/resolver/view/graph_controls.test.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Simulator } from '../test_utilities/simulator';
+import { noAncestorsTwoChildren } from '../data_access_layer/mocks/no_ancestors_two_children';
+import { nudgeAnimationDuration } from '../store/camera/scaling_constants';
+import '../test_utilities/extend_jest';
+
+describe('graph controls', () => {
+  let simulator: Simulator;
+  let originEntityID: string;
+  const resolverComponentInstanceID = 'graph-controls-test';
+
+  beforeEach(async () => {
+    const {
+      metadata: { databaseDocumentID, entityIDs },
+      dataAccessLayer,
+    } = noAncestorsTwoChildren();
+
+    simulator = new Simulator({
+      dataAccessLayer,
+      databaseDocumentID,
+      resolverComponentInstanceID,
+    });
+    originEntityID = entityIDs.origin;
+  });
+
+  it('should load graph controls', () => {
+    expect(simulator.graphControlElement().length).toBe(1);
+  });
+
+  describe('panning', () => {
+    const originalPositionStyle = { left: '746.93132px', top: '535.5792px' };
+    it('should pan west', async () => {
+      const originNode = simulator.processNodeElements({ entityID: originEntityID });
+      expect(originNode.getDOMNode()).toHaveStyle(originalPositionStyle);
+
+      const westPanButton = simulator.graphControlElement().find('[data-test-subj="west-button"]');
+      westPanButton.simulate('click');
+      simulator.runAnimationFramesTimeFromNow(nudgeAnimationDuration);
+
+      expect(originNode.getDOMNode()).toHaveStyle({ left: '796.93132px', top: '535.5792px' });
+    });
+
+    it('should pan south', async () => {
```

Comment:
could you phrase the spec like:
```
describe('when the user clicks the south panning button', () => {
  beforeEach(/* the click code does here*/)
  it('should show the origin node lower on the screen', () => {
    // the expect here
```

This way we can easily add more `it` blocks that have the same setup and we can easily tell which code sets up the condition vs which code asserts stuff

Reply:
Yea, this works really well. Thanks for the tip 👍